src/hooks/useCalculator.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import {
+    caculateNumber,
+    isOperator,
+    getCurrentDigits,
+    OperationType,
+    OperatorType
+} from '../utils';
+
+
+export const useCalculator = () => {
+    const [currentNumber, setCurrentNumber] = useState<number>(0);
+    const [operation, setOperation] = useState<string>('');
+    const [result, setResult] = useState<string>('0');
+
+    const [firstNumber, setFirstNumber] = useState<number>(0);
+
+
+    // 1.  들어오는 Digit에따라 숫자를 설정하고  result컴포넌트에 결과를 보여준다 .
+    const insertDigit = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const currentDigitNumber =getCurrentDigits(currentNumber,e.currentTarget.value)
+        try {
+            if(currentDigitNumber.length>3){
+                throw new Error('최대 3자리까지 입력가능합니다.')
+            }
+            setCurrentNumber(parseInt(currentDigitNumber))
+            setResult(currentDigitNumber)
+        } catch (message) {
+            alert(message)
+        }
```

Comment:
```suggestion
        } catch (error) {
            alert(error.message)
        }
```
message에 명시해주신 에러 내용이 담겨있어서 이렇게 넣어줘야해요..!

Reply:
이부분에 타입을 어떻게 해야될지 모르겠습니다.

Reply:
슬랙에서 이야기 나눠봤는데 혹시 추가로 궁금하신 부분이 생기면 말씀해주세요!